### ISSUE TITLE
Macos shortcuts problem, FocusableActionDetector widget not working.

### ISSUE THREAD

**Issue by CreateChance** [flutter/flutter#109912]
## Steps to Reproduce

1. Execute `flutter run` on the code sample 
2. Press meta + keyC to activate shortcuts, **It works, looks fine.**
3. Click text field(no text input), and click outer place to make the text field lose focus
4. Press meta + keyC to activate shortcuts,  **It doesn't works now, weird.**
5. Then, shortcut never works, whatever what you do.

**Expected results:** 
After step 3, pressing meta + keyC to activate shortcuts still works.

**Actual results:** 
After step 3, pressing meta + keyC to activate shortcuts never works.

<details>
<summary>Flutter version</summary>

```bash
Flutter 3.0.5 • channel stable • https://github.com/flutter/flutter.git
Framework • revision f1875d570e (5 周前) • 2022-07-13 11:24:16 -0700
Engine • revision e85ea0e79c
Tools • Dart 2.17.6 • DevTools 2.12.2
```
</details>
<details>
<summary>Macos version</summary>

<img src="https://user-images.githubusercontent.com/9158995/185734873-3760dccb-2d5b-4670-9da5-43a4b9250d64.png" height="80" />

</details>
<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  static const String title = 'Shortcuts and Actions Demo';

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: title,
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: Shortcuts(
        manager: LoggingShortcutManager(),
        shortcuts: <LogicalKeySet, Intent>{
          LogicalKeySet(LogicalKeyboardKey.meta, LogicalKeyboardKey.keyC): const CopyIntent(),
        },
        child: FocusableActionDetector(
            autofocus: true,
            actions: <Type, Action<Intent>>{
              CopyIntent: CopyAction(),
            },
            child: const Scaffold(
              body: Center(
                child: SizedBox(
                    width: 500,
                    child: TextField()
                ),
              ),
            )
        ),
      ),
    );
  }
}

class LoggingShortcutManager extends ShortcutManager {
  @override
  KeyEventResult handleKeypress(BuildContext context, RawKeyEvent event) {
    final KeyEventResult result = super.handleKeypress(context, event);
    print('Capture keyboard event.');
    if (result == KeyEventResult.handled) {
      print('Handle keyboard event.');
    }
    return result;
  }
}

class CopyIntent extends Intent {
  const CopyIntent();
}

class CopyAction extends Action<CopyIntent> {
  CopyAction();

  @override
  Object? invoke(covariant CopyIntent intent) {
    return null;
  }
}

```

</details>

<details>
  <summary>Logs</summary>

```
```

```
```

```
```

</details>

https://user-images.githubusercontent.com/9158995/185734579-de320f60-aa93-44d0-acc0-c2a9a7770f4e.mov

**Comment by CreateChance** [comment]
The problem is killing me, I need help.

**Comment by exaby73** [comment]
Hello @CreateChance. The problem you are facing is that when you click on the background, the focus is not given back to the `FocusableActionDetector`. I have modified your code to include some `FocusNode`s that manually request focus for it when the `TextField` unfocuses.

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

void main() => runApp(const MyApp());

class MyApp extends StatefulWidget {
  const MyApp({super.key});

  static const String title = 'Shortcuts and Actions Demo';

  @override
  State<MyApp> createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  final actionsFocusNode = FocusNode();
  final textFieldFocusNode = FocusNode();

  @override
  void initState() {
    super.initState();

    textFieldFocusNode.addListener(() {
      if (!textFieldFocusNode.hasFocus) {
        actionsFocusNode.requestFocus();
      }
    });
  }

  @override
  void dispose() {
    actionsFocusNode.dispose();
    actionsFocusNode.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: MyApp.title,
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: Shortcuts(
        manager: LoggingShortcutManager(),
        shortcuts: <LogicalKeySet, Intent>{
          LogicalKeySet(LogicalKeyboardKey.meta, LogicalKeyboardKey.keyC):
              const CopyIntent(),
        },
        child: FocusableActionDetector(
          autofocus: true,
          focusNode: actionsFocusNode,
          actions: <Type, Action<Intent>>{
            CopyIntent: CopyAction(),
          },
          child: Scaffold(
            body: Center(
              child: SizedBox(
                width: 500,
                child: TextField(
                  focusNode: textFieldFocusNode,
                ),
              ),
            ),
          ),
        ),
      ),
    );
  }
}

class LoggingShortcutManager extends ShortcutManager {
  @override
  KeyEventResult handleKeypress(BuildContext context, RawKeyEvent event) {
    final KeyEventResult result = super.handleKeypress(context, event);
    print('Capture keyboard event.');
    if (result == KeyEventResult.handled) {
      print('Handle keyboard event.');
    }
    return result;
  }
}

class CopyIntent extends Intent {
  const CopyIntent();
}

class CopyAction extends Action<CopyIntent> {
  CopyAction();

  @override
  Object? invoke(covariant CopyIntent intent) {
    return null;
  }
}
```

</details>

If you have any further questions, I recommend sites like StackOverflow or you can find additional resources on https://flutter.dev/community. Since this is not an issue with Flutter itself, I am closing this issue. Thank you.

**Comment by CreateChance** [comment]
Thanks for response, but one more problem here. After input chinese text in the text field, shortcuts not working again, but If you input english text, shortcuts still works.

**Comment by exaby73** [comment]
This looks like a separate issue. You can raise another issue for this but I am keeping this one closed. Thank you
